### ISSUE TITLE
IOS 17

### ISSUE THREAD

**Issue by SANSA27** [mspvirajpatel/Xcode_Developer_Disk_Images#48]
When can we get the developer disk image for ios 17 version ?

**Comment by SumitDixit1995** [comment]
When can we get the developer disk image for ios 17 version ?

**Comment by adhadimas91** [comment]
When can we get the developer disk image for ios 17 version ?

**Comment by Valkhes** [comment]
Hello, saw that nobody answered.
There won't be a disk image for ios 17. There is a new mechanism, named "Personalized developer disk image".
It's not just a file anymore, there is a whole process to mount developer disk image now for ios 17 and further.  Quick explanation would be that now there is a developer disk image created specifically for your device. There is also no lockdownd anymore, as a new service has been made for that.

I suggest to take a look at pymobiledevice3 ( https://github.com/doronz88/pymobiledevice3 ) which implemented most of the mechanisms, and which is able to mount the new personalized developer images.
I think libimobiledevice has surely managed to do it too now, but it was a bit late as pymobiledevice3 was faster to implement everything.

**Comment by SANSA27** [comment]
> Hello, saw that nobody answered. There won't be a disk image for ios 17. There is a new mechanism, named "Personalized developer disk image". It's not just a file anymore, there is a whole process to mount developer disk image now for ios 17 and further. Quick explanation would be that now there is a developer disk image created specifically for your device. There is also no lockdownd anymore, as a new service has been made for that.
> 
> I suggest to take a look at pymobiledevice3 ( https://github.com/doronz88/pymobiledevice3 ) which implemented most of the mechanisms, and which is able to mount the new personalized developer images. I think libimobiledevice has surely managed to do it too now, but it was a bit late as pymobiledevice3 was faster to implement everything.

Does the pymobiledevice provide the .dmg file as well as .signature file as well , I don't know how this pymobiledevice library works

**Comment by Valkhes** [comment]
> Does the pymobiledevice provide the .dmg file as well as .signature file as well , I don't know how this pymobiledevice library works

Yes, when installing pymobiledevice3 you can try to use the CLI command "pymobiledevice3 mounter auto-mount" for exemple. You will see in your home a new folder .pymobiledevice containing a .dmg and the .signature file. These are retrieved from a maintained repo by doronz88 (the creator of pymobiledevice)
You can find how everything is working directly in the pymobiledevice3 repo.

This .dmg and .signature can also be found in Xcode 15, in this folder : 
Xcode.app/Contents/Resources/CoreDeviceDDIs/iOS_DDI.dmg 

You can open the .dmg, and you will find inside the .signature and the .dmg.

I think there may be a new .dmg and .signature with new xcode version, but that's unsure for the moment.

EDIT : In fact, that's not a .dmg and a .signature anymore.
It's 3 files now that are necessary for the whole project : 
the .dmg, a .cache, and a BuildManifest

**Comment by SANSA27** [comment]
I have used pymobiledevice library to mount the developer disk image for ios 17.0.3 but even after mounting the developer disk image I can't be able to test the applications, but with the previous DDI's released by xcode when I mount the DDI I can be able to test the applications... Do you have any idea about that?? or any suggestions ?

**Comment by Valkhes** [comment]
> I have used pymobiledevice library to mount the developer disk image for ios 17.0.3 but even after mounting the developer disk image I can't be able to test the applications, but with the previous DDI's released by xcode when I mount the DDI I can be able to test the applications... Do you have any idea about that?? or any suggestions ?

Unfortunatly it's still very hard to debug for me, I won't be a great help for this.

**Comment by chirag05k** [comment]
**For those encountering problems running on iOS 17 devices with Xcode 14.x, there's a workaround mentioned in this Stack Overflow discussion:** [How to run on iOS 17 device using Xcode 14](https://stackoverflow.com/questions/76412754/how-to-run-on-ios-17-device-using-xcode-14/77388530#77388530).

**Comment by Montivillon** [comment]
> > I have used pymobiledevice library to mount the developer disk image for ios 17.0.3 but even after mounting the developer disk image I can't be able to test the applications, but with the previous DDI's released by xcode when I mount the DDI I can be able to test the applications... Do you have any idea about that?? or any suggestions ?
> 
> Unfortunatly it's still very hard to debug for me, I won't be a great help for this.

yo, do you know how to get that developer disk image for my device using pymobiledevice3? i need to use it for jitterbug

**Comment by pwneddesal** [comment]
> pymobiledevice3 mounter auto-mount

there is no signature file here 
![image](https://github.com/mspvirajpatel/Xcode_Developer_Disk_Images/assets/6646675/9a58bc89-5d7c-4d08-8cde-b090cf3bedd7)

**Comment by yanhuajun** [comment]
你好，你的来信已收到。

**Comment by guneesh27** [comment]
I have been able to fix this and future cases by downloading latest version of xcode, installing developer tools. 
The directory /Library/Developer/DeveloperDiskImages/iOS_DDI.dmg contains the DDI files which can be extracted and installed.

**Comment by yanhuajun** [comment]
你好，你的来信已收到。
